src/main/java/org/broadinstitute/consent/http/service/DataAccessRequestService.java

```diff
@@ -252,6 +253,11 @@ public DataAccessRequest createDataAccessRequest(User user, DataAccessRequest da
         if (Objects.isNull(user) || Objects.isNull(dataAccessRequest) || Objects.isNull(dataAccessRequest.getReferenceId()) || Objects.isNull(dataAccessRequest.getData())) {
             throw new IllegalArgumentException("User and DataAccessRequest are required");
         }
+
+        if (Objects.isNull(user.getLibraryCards()) || user.getLibraryCards().isEmpty()) {
+            throw new IllegalArgumentException("User must have a library card.");
+        }
```

Comment:
While important to have here, I like to do validation at the resource level and fail fast when we know we can't complete the process. The fact that the request gets here at all indicates that we should have returned an error previously.